pwndbg/aglib/heap/heap.py

```diff
@@ -2,13 +2,17 @@
 
 from typing import Any
 
-import gdb
-
 
 class MemoryAllocator:
     """Heap abstraction layer."""
 
-    def breakpoint(self, event: str) -> gdb.Breakpoint:
+    # This function isn't actually implemented anywhere. It originally returned
+    # `gdb.Breakpoint`, but, in order to facilitate the port to aglib, that
+    # type association was removed. It should be put back as soon as the
+    # Debugger-agnostic API gains the ability to set breakpoints.
+    #
+    # TODO: Change `Any` to the Debugger-agnostic breakpoint type when it gets created
+    def breakpoint(self, event: str) -> Any:
```

Comment:
Shall we remove this func if its not implemented anywhere? What is its purpose? Why does a heap MemoryAllocator class needs a bp function?

Reply:
I don't know. I don't really know what the intent was behind that method, originally. I don't think there's any reason for us to keep it, personally.

Reply:
Moved to https://github.com/pwndbg/pwndbg/issues/2404